packages/core-contracts/contracts/proxy/BeaconProxy.sol

```diff
@@ -1,21 +1,38 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import "./ForwardingProxy.sol";
+import "./AbstractProxy.sol";
+import "./Beacon.sol";
+import "./BeaconStorage.sol";
 import "../common/CommonErrors.sol";
-import "../interfaces/IBeacon.sol";
-import "../ownership/OwnableMixin.sol";
 import "../utils/ContractUtil.sol";
 
-abstract contract BeaconProxy is ForwardingProxy {
-    // solhint-disable-next-line no-empty-blocks
-    function _getBeacon() internal view virtual returns (address) {}
+contract BeaconProxy is AbstractProxy, BeaconStorage, CommonErrors, ContractUtil {
+    constructor(address firstBeacon) {
```

Comment:
Constructors are ok (and I think necessary) in proxies. It is the only way to set the first implementation without adding a an upgradeability admin selector to the ABI, which we can't have because of the proxy transparency problem.